lib/package-managers/npm.js

```diff
@@ -8,34 +8,18 @@ exports.NPM = class {
     this.executableName = 'npm';
   }
 
-  install(packages, workingDirectory) {
-    if (!workingDirectory) workingDirectory = process.cwd();
-
+  install(packages = [], workingDirectory = process.cwd()) {
     return new Promise((resolve, reject) => {
-      let executablePath = this.getExecutablePath(workingDirectory);
-      let cmd = this.installCommand(executablePath, packages);
-
-      let installProcess = childProcess.exec(cmd, {cwd: workingDirectory}, (error, stdout, stderr) => {
-        if (error) {
-          reject(error);
-          return;
-        }
-
-        resolve();
-      });
-
-      installProcess.stdout.on('data', data => console.log(data));
+      childProcess.spawn(
+        this.getExecutablePath(workingDirectory),
+        ['i', ...packages],
```

Comment:
I think you still can normalise this line to a method, so yarn can override the args.

Reply:
Ok updated the code, changed the abbreviation to 'install' which works for Yarn. Checked the code and seems like `installCommand` is not used at all so I decided to also remove this from the Yarn class.